tests/services/assignment_service_test.py

```diff
@@ -0,0 +1,82 @@
+import unittest
+from unittest.mock import MagicMock
+from sqlalchemy.orm import Session
+from app.services import AssignmentService
+from app.models import AssignmentModel
+from app.core.exceptions import AssignmentNotFoundException
+from ..data.database.assignment import assignment_data
+
+class TestAssignmentService(unittest.IsolatedAsyncioTestCase):
+    async def asyncSetUp(self):
+        self.mock_session = MagicMock()
+        self.assignment_service = AssignmentService(session=self.mock_session)
+
+
+    async def test_get_assignment_by_id_success(self):
+        mock_assignment = assignment_data["available"]
+        self.mock_session.query().first.return_value = mock_assignment
+
+        result = await self.assignment_service.get_assignment_by_id(id=1)
```

Comment:
I don't like this paradigm of using outside data (from `assignment_data` above) and basically depending on it being of a certain state (some state referenced by id=1) for the test to work. This encourages us to use that state for multiple tests, which then causes problems if you change the state later because it could break certain tests but not others. I think each set of tests (so, each file/test class) needs to set up its own data for itself. That's the way I've always seen it done.